sphinx/lib/dh_sphinx.py

```diff
@@ -0,0 +1,96 @@
+
+import os
+import shutil
+import pkgutil
+
+def glob_package_names(packages):
+    rst = []
+
+    for package in packages:
+        rst.append(package.__name__)
+
+        if hasattr(package,"__path__"):
+            for importer, modname, ispkg in pkgutil.walk_packages(path=package.__path__, prefix=package.__name__+'.', onerror=lambda x: None):
+                rst.append(modname)
+
+    return rst
+
+
+def _add_package(tree, package):
+    n = package[0]
+
+    if n not in tree:
+        tree[n] = {}
+
+    if len(package) > 1:
+        _add_package(tree[n],package[1:])
+
+
+def package_tree(package_names):
+    rst = {}
+    for pn in package_names:
+        spn = pn.split('.')
+        _add_package(rst, spn)
+    return rst
+
+
+def make_rst_tree(package, tree):
+    package_name = ".".join(package)
+
+    if len(tree) == 0:
+        toctree = ""
+    else:
+        toctree = ".. toctree::\n"
+        for k in tree:
+            p = package.copy()
+            p.append(k)
+            pn = ".".join(p)
+            toctree += "%s%s <%s>\n"%(" "*4,k,pn)
+
+    rst = "%s\n%s\n\n%s\n.. automodule:: %s\n    :members:\n    :show-inheritance:\n    :special-members: __init__\n    :undoc-members:\n\n"%(package_name,"="*len(package_name),toctree,package_name)
```

Comment:
i think we should remove the  'special-members: __init__' option, at least for pydeephaven. I only want anything with a docstring to be documented, nothing else.